exporter/elasticsearchexporter/README.md

```diff
@@ -68,6 +68,9 @@ This exporter supports sending OpenTelemetry logs to [Elasticsearch](https://www
     - `ecs`: Try to map fields defined in the
              [OpenTelemetry Semantic Conventions](https://github.com/open-telemetry/semantic-conventions)
              to [Elastic Common Schema (ECS)](https://www.elastic.co/guide/en/ecs/current/index.html).
+    - `raw`: Omit the `Attributes.` string prefixed to field names for log and 
```

Comment:
Is it possible for a user to want both "ecs" and "raw" functionality? With this configuration, this is not possible.

Looking at the code, I'm not sure the "ecs" mapping mode actually does anything? Is there a functional difference between "none" and "ecs" mapping mode in the current state of the exporter?

Reply:
> Looking at the code, I'm not sure the "ecs" mapping mode actually does anything? Is there a functional difference between "none" and "ecs" mapping mode in the current state of the exporter?

Yeah, I don't see mapping mode being used at all (before this PR introduced `raw`).  So I don't think there's any functional difference between `none` and `ecs` mapping modes today.  

I looked at the PR that introduced the `ecs` mapping mode — https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2324 — but I don't see any explanation for why it is functionally the same as `raw`.  My guess is that `ecs` was introduced as part of some future-proofing work that was never done?  

Given that `ecs` mapping mode doesn't really do anything today, I can't say that users may want both `ecs` and `raw` functionality at the same time.  Perhaps we should deprecate `ecs` (in a separate PR) and later remove it, and then go ahead with this PR introducing `raw` mapping mode?  If so, are there any guidelines around such deprecation and removal (I'm new to the OTel community)?

Reply:
This makes sense. If the `ecs` mode does not do anything, then there's no point in enabling both `ecs` and `raw` at the same time.

>  Perhaps we should deprecate `ecs` (in a separate PR) and later remove it

Agree 💯 

> and then go ahead with this PR introducing raw mapping mode

I think both changes can be done independently, can't they? We can move on with introducing `raw` mode in this PR and separately create another PR to deprecate `ecs`. To put it yet another way, `ecs` removal does not block introduction of `raw` in my view. 

As to the deprecation and removal process, the `ecs` mode is currently the default, so I think the following steps make sense:

1. Deprecate the `ecs` mode and at the same time change the default from `ecs` to `none` - this is a no-op change assuming both modes are functionally the same.
2. Remove the `ecs` mode after a long time - perhaps 3 months, perhaps 6 months - I suppose there's no rush here, because having this option doesn't really hurt us, right?

I don't think introducing a feature gate for this removal makes sense, given that the fixing the configuration - removing `mode: ecs` or changing it to `mode: none` - is simpler than using a feature gate.

Reply:
Thanks @astencel-sumo, I've created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30441 to track the deprecation of `mode: ecs` and changing the default to `mode: none`.

I agree we can just go ahead with this PR here introducing `mode: raw` now itself, independently of what becomes of `mode: ecs`.